command.go

```diff
@@ -230,6 +231,33 @@ func (c *Command) HelpFunc() func(*Command, []string) {
 	}
 }
 
+// Output the usage for the command
+// Used when a user provides invalid input
+// Can be defined by user by overriding UsageFunc
+func (c *Command) Usage() error {
+	c.mergePersistentFlags()
+	err := tmpl(c.getOutOrStderr(), c.UsageTemplate(), c)
```

Comment:
@dnephin like in https://github.com/spf13/cobra/pull/305/commits/d6bf4ef243b7f3b96d8a4249e63af711ef33d53e